open_fdd/desktop/drivers/weather_driver.py

```
@@ -17,7 +21,45 @@ class WeatherFetchResult:
+            start = end - timedelta(days=max(1, int(days_back)))
+            base_url = os.getenv("OFDD_OPEN_METEO_BASE_URL", "https://archive-api.open-meteo.com/v1/archive").strip()
+            params = {
+                "latitude": float(lat),
+                "longitude": float(lon),
+                "start_date": start.date().isoformat(),
+                "end_date": end.date().isoformat(),
+                "hourly": "temperature_2m,relative_humidity_2m",
+                "timezone": os.getenv("OFDD_OPEN_METEO_TIMEZONE", "UTC").strip() or "UTC",
+            }
+            url = f"{base_url}?{urllib.parse.urlencode(params)}"
+            req = urllib.request.Request(url, method="GET")
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)